### PR TITLE
first commit (draft) add amplify

### LINKED ISSUE #52: Add AMPLify to pipeline
### Description of feature

Add AMPlify to pipeline now it is in modules

### PR DESCRIPTION
**Pull request by louperelo**
closes #52

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/amp.nf [hunk 1]
```diff
@@ -14,8 +15,15 @@ workflow AMP {
     ch_versions = Channel.empty()
     ch_mqc      = Channel.empty()
 
-    // TODO AMPEP(?)
+
     // TODO ampir
+    
+/*
```
Comment (jfy133) [inline subworkflows/local/amp.nf:21]:
```suggestion
ch_faa_for_amplify = faa
ch_faa_for_hmmsearch = faa
```

File: subworkflows/local/amp.nf [hunk 2]
```diff
@@ -14,8 +15,15 @@ workflow AMP {
...
+    if ( !params.amp_skip_amplify ) {
+        AMPLIFY_PREDICT ( faa, [] )
+        ch_versions = ch_versions.mix(AMPLIFY_PREDICT.out.versions)
+    }
+*/
```
Comment (jfy133) [inline subworkflows/local/amp.nf:26]:
```suggestion

```

File: subworkflows/local/amp.nf [hunk 3]
```diff
@@ -14,8 +15,15 @@ workflow AMP {
     ch_versions = Channel.empty()
     ch_mqc      = Channel.empty()
 
-    // TODO AMPEP(?)
+
     // TODO ampir
+    
+/*
+    if ( !params.amp_skip_amplify ) {
+        AMPLIFY_PREDICT ( faa, [] )
```
Comment (jfy133) [inline subworkflows/local/amp.nf:23]:
```suggestion
        AMPLIFY_PREDICT ( ch_faa_for_amplify, [] )
```

File: workflows/funcscan.nf [hunk 4]
```diff
@@ -108,7 +108,7 @@ workflow FUNCSCAN {
     if ( params.run_amp_screening ) {
 
         if ( !params.amp_skip_hmmsearch ) {
-            AMP ( ch_prepped_input, PROKKA.out.faa )
+            AMP ( ch_prepped_input, PROKKA.out.faa.dump(tag: 'faa') )
```
Comment (jfy133) [inline workflows/funcscan.nf:111]:
```suggestion
            AMP ( ch_prepped_input, PROKKA.out.faa )
```

**Review by jfy133 — COMMENTED** [review]
Our fixes

File: CITATIONS.md [hunk 5]
```diff
@@ -30,8 +30,12 @@
   > Arango-Argoty, Gustavo, Emily Garner, Amy Pruden, Lenwood S. Heath, Peter Vikesland, and Liqing Zhang. 2018. "DeepARG: A Deep Learning Approach for Predicting Antibiotic Resistance Genes from Metagenomic Data." Microbiome 6 (1): 23.
 
 * [hmmsearch](https://doi.org/10.1371/journal.pcbi.1002195.)
+
   > Eddy, Sean R. 2011. “Accelerated Profile HMM Searches.” PLoS Computational Biology 7 (10): e1002195.
 
+* [AMPlify](https://doi.org/10.1186/s12864-022-08310-4)
+  > Chenkai Li, Darcy Sutherland, S. Austin Hammond, Chen Yang, Figali Taho, Lauren Bergman, Simon Houston, René L. Warren, Titus Wong, Linda M. N. Hoang, Caroline E. Cameron, Caren C. Helbing & Inanc Birol. 2022. "AMPlify: attentive deep learning model for discovery of novel antimicrobial peptides effective against WHO priority pathogens". BMC Genomics 23, 77.
```
Comment (jfy133) [inline CITATIONS.md:37]:
```suggestion

  > Chenkai Li, Darcy Sutherland, S. Austin Hammond, Chen Yang, Figali Taho, Lauren Bergman, Simon Houston, René L. Warren, Titus Wong, Linda M. N. Hoang, Caroline E. Cameron, Caren C. Helbing & Inanc Birol. 2022. "AMPlify: attentive deep learning model for discovery of novel antimicrobial peptides effective against WHO priority pathogens". BMC Genomics 23, 77.
```

File: docs/output.md [hunk 6]
```diff
@@ -278,19 +279,16 @@ The `*.ARG` output files contain the following fields:
...
-<!--### AMPlify
+### AMPlify
 
 <details markdown="1">
 <summary>Output files</summary>
 
-* `amplify/`
-    * `output1`: xxx
-    * `output2/`: xxx
+-_`amplify/` -_ `AMPlify_results.tsv`: contig amino-acid sequences with prediction result (AMP or non-AMP) and information on sequence length, charge, probability score, AMPlify log-scaled score)
```
Comment (jfy133) [inline docs/output.md:287]:
```suggestion
-_`amplify/` -_ `*_results.tsv`: contig amino-acid sequences with prediction result (AMP or non-AMP) and information on sequence length, charge, probability score, AMPlify log-scaled score)
```

File: nextflow_schema.json [hunk 7]
```diff
@@ -405,6 +416,9 @@
         {
             "$ref": "#/definitions/amp_hmmer_hmmsearch"
         },
+        {
+            "$ref": "#/definitions/new_group_1"
```
Comment (jfy133) [inline nextflow_schema.json:420]:
```suggestion
            "$ref": "#/definitions/amp_amplify_predict"
```

File: nextflow_schema.json [hunk 8]
```diff
@@ -167,6 +167,17 @@
                 }
             }
         },
+        "new_group_1": {
+            "title": "New Group 1",
```
Comment (jfy133) [inline nextflow_schema.json:171]:
```suggestion
            "title": "AMP-AMPlify-Predict",
```

**Review by jfy133 — APPROVED** [review]
Couple of monior things but otherwise looks good to me :) thakns @louperelo

File: nextflow_schema.json (hunk 8 above)
Comment (louperelo) [reply nextflow_schema.json]: Where did this "New Group 1" came from? Somehow these changes in the nextflow_schema.json made the checks fail ...

Comment (jfy133) [reply nextflow_schema.json]: I think the group name didn't get saved when you changed it in the builder

**Pull request merged**